Framework/Algorithms/src/CalMuonDetectorPhases.cpp

```diff
@@ -161,30 +167,52 @@ void CalMuonDetectorPhases::fitWorkspace(const API::MatrixWorkspace_sptr &ws,
   const static std::string success = "success";
   for (int wsIndex = 0; wsIndex < nhist; wsIndex++) {
     reportProgress(wsIndex, nhist);
-    auto fit = createChildAlgorithm("Fit");
-    fit->initialize();
-    fit->setPropertyValue("Function", funcStr);
-    fit->setProperty("InputWorkspace", ws);
-    fit->setProperty("WorkspaceIndex", wsIndex);
-    fit->setProperty("CreateOutput", true);
-    fit->setPropertyValue("Output", groupName);
-    fit->execute();
-
-    std::string status = fit->getProperty("OutputStatus");
-    if (!fit->isExecuted() || status != success) {
-      std::ostringstream error;
-      error << "Fit failed for spectrum at workspace index " << wsIndex;
-      error << ": " << status;
-      throw std::runtime_error(error.str());
-    }
+    auto yValues = ws->readY(wsIndex);
+    auto emptySpectrum = std::all_of(yValues.begin(), yValues.end(), isZero);
```

Comment:
Could use a lambda here:
```
std::all_of(yValues.begin(), yValues.end(), [](double value){ return value == 0.;} );
```
? It saves having to look elsewhere for the definition of such a small function